android-app/app/src/main/java/com/nimtome/app/CharacterDetailsActivity.kt

```diff
@@ -66,19 +68,19 @@ class CharacterDetailsActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
-            DndSpellsTheme(darkColors = colorPalette.darkColors,
-                lightColors = colorPalette.lightColors) {
-                // A surface container using the 'background' color from the theme
-                val viewModel = ViewModelProvider(this)
-                val nameInDb = intent.getStringExtra(KEY_NAME) ?: ""
-                val character =
-                    viewModel[CharacterViewModel::class.java].get(nameInDb).observeAsState().value
-                        ?: DndCharacter()
-                val spells =
-                    viewModel[SpellViewModel::class.java].allSpells.observeAsState().value
-                        ?: listOf()
+            val viewModel = ViewModelProvider(this)
+            val nameInDb = intent.getStringExtra(KEY_NAME) ?: ""
+            val character by viewModel[CharacterViewModel::class.java].get(nameInDb).observeAsState()
+            val preferredColorPalette = character?.preferredColorPalette ?: colorPalette
```

Comment:
Could you import the class and then use its comp, object ? the direct import seems weird